code/__DEFINES/math.dm

```diff
@@ -10,6 +10,25 @@
 #define T20C 293.15					// 20degC
 #define TCMB 2.7					// -270.3degC
 
-#define Clamp(x, y, z) 	(x <= y ? y : (x >= z ? z : x))
-
-#define CLAMP01(x) 		(Clamp(x, 0, 1))
+#define Clamp(x, y, z)			 	((x) <= (y) ? (y) : ((x) >= (z) ? (z) : (x)))
+#define CLAMP01(x) 					(Clamp((x), 0, 1))
+#define SIMPLE_SIGN(X)				((X) < 0 ? -1 : 1)
+#define SIGN(X)						((X) ? SIMPLE_SIGN(X) : 0)
+#define hypotenuse(Ax, Ay, Bx, By)	(sqrt(((Ax) - (Bx))**2 + ((Ay) - (By))**2))
+#define Ceiling(x)					(-round(-(x)))
+#define Tan(x)						(sin(x) / cos(x))
+#define Cot(x)						(1 / Tan(x))
+#define Csc(x)						(1 / sin(x))
+#define Sec(x)						(1 / cos(x))
+#define Default(a, b)				((a) ? (a) : (b))
```

Comment:
```#define Default(a, b) (a || b)```
if ```a``` is null it'll pick ```b```

this is because ```||``` works as a null-coalescencing operator